packages/typespec-test/test/lro_modular/generated/typespec-ts/src/rest/pollingHelper.ts

```diff
@@ -0,0 +1,103 @@
+  SimplePollerLike,
+  createHttpPoller,
+} from "@azure/core-lro";
+import {
+  CreateOrReplace200Response,
+  CreateOrReplace201Response,
+  CreateOrReplaceDefaultResponse,
+  CreateOrReplaceLogicalResponse,
+  Export202Response,
+  ExportDefaultResponse,
+  ExportLogicalResponse,
+} from "./responses";
+/**
+ * Helper function that builds a Poller object to help polling a long running operation.
+ * @param client - Client to use for sending the request to get additional pages.
+ * @param initialResponse - The initial response.
+ * @param options - Options to set a resume state or custom polling interval.
+ * @returns - A poller object to poll for operation state updates and eventually get the final response.
+ */
+export async function getLongRunningPoller<
+  TResult extends ExportLogicalResponse | ExportDefaultResponse
+>(
+  client: Client,
+  initialResponse: Export202Response | ExportDefaultResponse,
+  options?: CreateHttpPollerOptions<TResult, OperationState<TResult>>
+): Promise<SimplePollerLike<OperationState<TResult>, TResult>>;
+export async function getLongRunningPoller<
+  TResult extends
+    | CreateOrReplaceLogicalResponse
+    | CreateOrReplaceDefaultResponse
+>(
+  client: Client,
+  initialResponse:
+    | CreateOrReplace200Response
+    | CreateOrReplace201Response
+    | CreateOrReplaceDefaultResponse,
+  options?: CreateHttpPollerOptions<TResult, OperationState<TResult>>
+): Promise<SimplePollerLike<OperationState<TResult>, TResult>>;
+export async function getLongRunningPoller<TResult extends HttpResponse>(
+  client: Client,
+  initialResponse: TResult,
+  options: CreateHttpPollerOptions<TResult, OperationState<TResult>> = {}
+): Promise<SimplePollerLike<OperationState<TResult>, TResult>> {
+  const poller: LongRunningOperation<TResult> = {
+    requestMethod: initialResponse.request.method,
+    requestPath: initialResponse.request.url,
+    sendInitialRequest: async () => {
+      // In the case of Rest Clients we are building the LRO poller object from a response that's the reason
+      // we are not triggering the initial request here, just extracting the information from the
+      // response we were provided.
+      return getLroResponse(initialResponse);
+    },
+    sendPollRequest: async (path) => {
+      // This is the callback that is going to be called to poll the service
+      // to get the latest status. We use the client provided and the polling path
+      // which is an opaque URL provided by caller, the service sends this in one of the following headers: operation-location, azure-asyncoperation or location
+      // depending on the lro pattern that the service implements. If non is provided we default to the initial path.
+      const response = await client
+        .pathUnchecked(path ?? initialResponse.request.url)
+        .get();
+      const lroResponse = getLroResponse(response as TResult);
+      lroResponse.rawResponse.headers["x-ms-original-url"] =
+        initialResponse.request.url;
```

Comment:
I don't understand this line, could you please explain?

Reply:
I generate this by referrng the RLC helper function and notice that we have below header set:
https://github.com/Azure/autorest.typescript/blob/648ae3d5ced12dcb874b93f9bb17f6738d8aea5e/packages/rlc-common/src/static/pollingContent.ts#L75-L76